### PR TITLE
Add SiCK-PAD Keyboard

### PR DESCRIPTION
**Pull request by jmschneider**
## Add SiCK-PAD Keyboard

Add a keyboad config for the [SiCK-PAD handwired numpad](https://www.thingiverse.com/thing:3682168)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/sick_pad/config.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#ifndef CONFIG_H
+#define CONFIG_H
```
Comment (ridingqwerty) [inline keyboards/handwired/sick_pad/config.h:2]:
```suggestion
#pragma once
```

Prefer `#pragma once` to these old-style include guards.

File: keyboards/handwired/sick_pad/config.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (ridingqwerty) [inline keyboards/handwired/sick_pad/config.h:33]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/handwired/sick_pad/config.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
+
```
Comment (ridingqwerty) [inline keyboards/handwired/sick_pad/config.h:48]:
```suggestion
```

These are core settings.

File: keyboards/handwired/sick_pad/config.h [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 0
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#endif
+
+#endif
```
Comment (ridingqwerty) [inline keyboards/handwired/sick_pad/config.h:58]:
```suggestion
```

Use of `#pragma once` above obviates the need for this closing `#endif`.

File: keyboards/handwired/sick_pad/kb.c [hunk 5]
```diff
@@ -0,0 +1 @@
+#include "kb.h"
```
Comment (ridingqwerty) [inline keyboards/handwired/sick_pad/kb.c:1]:
```suggestion
#include "sick_pad.h"
```

Please change this header to `#include "sick_pad.h"` and rename this file to "sick_pad.c".

File: keyboards/handwired/sick_pad/kb.h [hunk 6]
```diff
@@ -0,0 +1,20 @@
+#ifndef KB_H
+#define KB_H
```
Comment (ridingqwerty) [inline keyboards/handwired/sick_pad/kb.h:2]:
```suggestion
#pragma once
```

Prefer `#pragma once` to these old style include guards.  Also, please rename this file to "sick_pad.h".

File: keyboards/handwired/sick_pad/kb.h [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+	     K41, K42, K43  \
+) { \
+	{ K00,   K01,   K02,   K03 }, \
+	{ K10,   K11,   K12,   KC_NO }, \
+	{ K20,   K21,   K22,   K23 }, \
+	{ K30,   K31,   K32,   KC_NO }, \
+	{ KC_NO, K41,   K42,   K43 }  \
+}
+
+#endif
```
Comment (ridingqwerty) [inline keyboards/handwired/sick_pad/kb.h:20]:
```suggestion
```

`#pragma once` above obviates the need for this closing `#endif`.

File: keyboards/handwired/sick_pad/kb.h [hunk 8]
```diff
@@ -0,0 +1,20 @@
+#ifndef KB_H
+#define KB_H
+
+#include "quantum.h"
+
+#define KEYMAP( \
```
Comment (ridingqwerty) [inline keyboards/handwired/sick_pad/kb.h:6]:
```suggestion
#define LAYOUT( \
```

File: keyboards/handwired/sick_pad/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,65 @@
+#include "kb.h"
```
Comment (ridingqwerty) [inline keyboards/handwired/sick_pad/keymaps/default/keymap.c:1]:
```suggestion
#include QMK_KEYBOARD_H
```

Prefer the more generic `#include QMK_KEYBOARD_H` here.

File: keyboards/handwired/sick_pad/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+    KEYMAP(KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+    KEYMAP(KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+    KEYMAP(KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+    KEYMAP(KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+    KEYMAP(KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS)
```
Comment (ridingqwerty) [inline keyboards/handwired/sick_pad/keymaps/default/keymap.c:35]:
```suggestion
    [0] = LAYOUT(
        KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,   KC_P8,   KC_P9,
        KC_P4,   KC_P5,   KC_P6, KC_PPLS,
        KC_P1,   KC_P2,   KC_P3,
                 KC_P0,   KC_PDOT, KC_PENT
    )
```

Prefer `LAYOUT` to `KEYMAP` for layout macro name, and remove unreachable layers (plus, this many layers will massively bloat the firmware unnecessarily).

File: keyboards/handwired/sick_pad/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    if (usb_led & (1 << USB_LED_COMPOSE)) {
+    } else {
+    }
+
+    if (usb_led & (1 << USB_LED_KANA)) {
+    } else {
+    }
+}
```
Comment (ridingqwerty) [inline keyboards/handwired/sick_pad/keymaps/default/keymap.c:65]:
```suggestion
```

This can all be removed as it appears to be unused.

**Review by ridingqwerty — CHANGES_REQUESTED** [review]
There are some older conventions in use here.

If you commit my suggestions and rename the `kb.{c,h}` files to `sick_pad.{c,h}`, you should be good to go.

**Comment by jmschneider** [comment]
Thank you for the guidance @ridingqwerty, I am new to working with QMK. I think I've got every thing set now. Let me know if there are any other blockers.

**Review by ridingqwerty — APPROVED** [review]
Looking good.

File: keyboards/handwired/sick_pad/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,11 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
```
Comment (zvecr) [inline keyboards/handwired/sick_pad/keymaps/default/keymap.c:4]:
```suggestion
    [0] = LAYOUT_numpad_5x4(
```

File: keyboards/handwired/sick_pad/sick_pad.h [hunk 13]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT( \
```
Comment (zvecr) [inline keyboards/handwired/sick_pad/sick_pad.h:5]:
```suggestion
#define LAYOUT_numpad_5x4( \
```

File: keyboards/handwired/sick_pad/rules.mk [hunk 14]
```diff
@@ -0,0 +1,2 @@
+# MCU name
+MCU = STM32F303
```
Comment (zvecr) [inline keyboards/handwired/sick_pad/rules.mk:2]:
```suggestion
MCU = STM32F303

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output

LAYOUTS = numpad_5x4
```

File: keyboards/handwired/sick_pad/config.h [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    qmkbuilder
+#define PRODUCT         keyboard
+#define DESCRIPTION     Keyboard
+
```
Comment (zvecr) [inline keyboards/handwired/sick_pad/config.h:12]:
```suggestion

```

File: keyboards/handwired/sick_pad/config.h [hunk 16]
```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    qmkbuilder
+#define PRODUCT         keyboard
```
Comment (zvecr) [inline keyboards/handwired/sick_pad/config.h:10]: Something more relevant here (not sure what though)

File: keyboards/handwired/sick_pad/config.h [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
+
```
Comment (zvecr) [inline keyboards/handwired/sick_pad/config.h:30]:
```suggestion
```

File: keyboards/handwired/sick_pad/config.h [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 0
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#endif
```
Comment (zvecr) [inline keyboards/handwired/sick_pad/config.h:46]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/handwired/sick_pad/config.h [hunk 19]
```diff
@@ -0,0 +1,46 @@
+#pragma once
```
Comment (noroadsleft) [inline keyboards/handwired/sick_pad/config.h:1]: Could add a license header here, please?

Here's a sample:

https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/sick_pad/config.h (hunk 16 above)
Comment (noroadsleft) [inline keyboards/handwired/sick_pad/config.h:10]:
```suggestion
#define MANUFACTURER    FedorSosnin
#define PRODUCT         SiCK-PAD
```

File: keyboards/handwired/sick_pad/config.h [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x6060
```
Comment (noroadsleft) [inline keyboards/handwired/sick_pad/config.h:7]:
```suggestion
#define PRODUCT_ID      0xDA20
```

File: keyboards/handwired/sick_pad/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,11 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline keyboards/handwired/sick_pad/keymaps/default/keymap.c:1]: Could add a license header here, please?

File: keyboards/handwired/sick_pad/readme.md [hunk 22]
```diff
@@ -0,0 +1,15 @@
+# SiCK-PAD
+
+![SiCK-PAD](https://cdn.thingiverse.com/assets/79/f6/22/c5/93/IMG_7816.JPG)
+
+The SiCK-PAD is a custom 3D printed mechanical numpad built from scratch without the price tag often associated with one.
+
+-   Keyboard Maintainer: [umbynos](https://github.com/umbynos)
```
Comment (noroadsleft) [inline keyboards/handwired/sick_pad/readme.md:7]:
```suggestion
-   Keyboard Maintainer: [jmschneider](https://github.com/jmschneider)
```

File: keyboards/handwired/sick_pad/sick_pad.c [hunk 23]
```diff
@@ -0,0 +1 @@
+#include "sick_pad.h"
```
Comment (noroadsleft) [inline keyboards/handwired/sick_pad/sick_pad.c:1]: Could add a license header here, please?

File: keyboards/handwired/sick_pad/sick_pad.h [hunk 24]
```diff
@@ -0,0 +1,17 @@
+#pragma once
```
Comment (noroadsleft) [inline keyboards/handwired/sick_pad/sick_pad.h:1]: Could add a license header here, please?

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/10429/files).

File: keyboards/handwired/sick_pad/config.h (hunk 20 above)
Comment (jmschneider) [reply keyboards/handwired/sick_pad/config.h]: Is there any particular for using `0xDA20`?

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**